File: c/conditionals.c
/*
CONDITIONAL STATEMENTS IN C
This file demonstrates conditional statements in C:
- if statement
- if-else statement
- else-if ladder
- nested if
- switch case statement
*/

#include <stdio.h>

int main() {
    int num = 10;
    
    // Simple if statement
    if (num > 0) {
        printf("%d is positive\n\n", num);
    }
    
    // if-else statement
    if (num % 2 == 0) {
        printf("%d is even\n\n", num);
    } else {
        printf("%d is odd\n\n", num);
    }
    
    // else-if ladder
    int marks = 75;
    printf("Grade: ");
    if (marks >= 90) {
        printf("A\n\n");
    } else if (marks >= 80) {
        printf("B\n\n");
    } else if (marks >= 70) {
        printf("C\n\n");
    } else if (marks >= 60) {
        printf("D\n\n");
    } else {
        printf("F\n\n");
    }
    
    // Nested if
    int age = 25;
    if (age >= 18) {
        if (age <= 60) {
            printf("You are eligible to work\n\n");
        } else {
            printf("You are too old to work\n\n");
        }
    } else {
        printf("You are too young to work\n\n");
    }
    
    // Switch case
    char grade = 'B';
    printf("Grade comments: ");
    switch (grade) {
        case 'A':
            printf("Excellent!\n");
            break;
        case 'B':
            printf("Good job!\n");
            break;
        case 'C':
            printf("You can do better\n");
            break;
        case 'D':
            printf("Needs improvement\n");
            break;
        case 'F':
            printf("Failed\n");
            break;
        default:
            printf("Invalid grade\n");
    }
    
    return 0;
}

File: python/conditionals.py
"""
PYTHON CONDITIONAL STATEMENTS
This file demonstrates conditional statements in Python:
- if statement
- if-else statement
- elif ladder
- nested if
- ternary operator
- match-case (Python 3.10+)
"""

# Simple if statement
num = 10
if num > 0:
    print(f"{num} is positive\n")

# if-else statement
if num % 2 == 0:
    print(f"{num} is even\n")
else:
    print(f"{num} is odd\n")

# elif ladder
marks = 75
print("Grade: ", end="")
if marks >= 90:
    print("A\n")
elif marks >= 80:
    print("B\n")
elif marks >= 70:
    print("C\n")
elif marks >= 60:
    print("D\n")
else:
    print("F\n")

# Nested if
age = 25
if age >= 18:
    if age <= 60:
        print("You are eligible to work\n")
    else:
        print("You are too old to work\n")
else:
    print("You are too young to work\n")

# Ternary operator
max_num = a if (a := 10) > (b := 5) else b
print(f"Maximum between {a} and {b} is {max_num}\n")

# Match-case (Python 3.10+)
grade = 'B'
print("Grade comments: ", end="")
match grade:
    case 'A':
        print("Excellent!")
    case 'B':
        print("Good job!")
    case 'C':
        print("You can do better")
    case 'D':
        print("Needs improvement")
    case 'F':
        print("Failed")
    case _:
        print("Invalid grade")